### DISCUSSION TITLE
How to speed up forge?

### DISCUSSION THREAD

**Discussion by a3nima** [lllyasviel/stable-diffusion-webui-forge#1762] [Q&A]
System: Windows 10; Nvidia 3090 (24GB VRAM); 64GB Sys RAM; OS+Forge running on M2 SSDs.

Comfy (SwarmUI) works way faster for me with Flux than forge (~30%).

What settings and start parameters should I use to increase the generation speed in forge?

Currently I'm using:
 python: 3.10.6  •  torch: 2.3.1+cu121 and no xformers, as there where problems reported with Flux in the beginning.
Also I haven't set any other start parameters..

What settings would you suggest, also in the UI (GPU weights) when generating with Forge and Flux?

Thanks!

**Comment by lllyasviel** [comment]
i just tested my 3090 with win10 and forge is still 6 or 5 seconds faster than them in first pass with model load; and about 2 or 3 seconds faster than them in second pass when prompt cached.

however, if you have full console log of both software to verify your claim, i will take a look

also you can click sdxl on the left-top 'ui' radio and then click flux to refresh gpu configurations to flux default, when collecting console logs and comparisons.

**Reply by aggregate15** [reply]
I will second this. Tried Comfy and Forge Side to side many times on win 11 and 10 and forge is on average, 15% faster on both my 4060 and 4080, respectively. Must be something to do with OP settings.

**Reply by a3nima** [reply]
I've already had my forge log posted in another thread [thread](https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/1050#discussioncomment-10442000) and there was nothing to be found in it, that would give us a clue. It's the same for me with the dev1 fp16 model. For Swarm/Comfy I dont know how to get a more detailed log. But it's not SwarmUI that I want to change.

I had hoped that maybe xformers or another pytorch version could help. 

Edit: comfy uses pytorch 2.4.0 maybe that's also responsible

How many seconds do you need for a 1024x1024 image using Flux Dev 1 fp16 with t5xxl/clipL/ae, 20 steps, Euler ?

**Reply by a3nima** [reply]
I think I will do a complete reinstall of forge and see if I can get more similar generation speeds.
